### PR TITLE
[testem-server] refactoring the server to remove any sync fs logic

### PR DESCRIPTION
**Pull request by gabrielcsapo**
Refactoring any of the sync calls being made to increase speed of the http files being served.

### CODE REVIEW

**Comment by stefanpenner** [comment]
async/await is not supported in node 6, which this project still supports until April 2019 as we follow nodes LTS policy: https://github.com/nodejs/Release#release-schedule

File: lib/server/index.js [hunk 1]
```diff
@@ -83,53 +103,52 @@ class Server extends EventEmitter {
...
+    app.get('/testem.js', asyncMiddleware(async (req, res) => {
+      await this.serveTestemClientJs(req, res);
+    }));
 
     app.all(/^\/(?:-?[0-9]+)(\/.+)$/, serveStaticFile);
     app.all(/^(.+)$/, serveStaticFile);
 
     app.use((err, req, res, next) => {
       if (err) {
+        console.log(err)
```
Comment (stefanpenner) [inline lib/server/index.js]: is this intentional, should this use `log` ?

**Comment by stefanpenner** [comment]
> Refactoring any of the sync calls being made to increase speed of the http files being served.

Do you have benchmarks demonstrating an improvement?

I am suspicious of this claim, since all IO was already async prior to this PR.

That being said, I am all for using native promises over polyfils.

**Comment by gabrielcsapo** [comment]
closing for now

**Pull request closed without merging**